### ISSUE TITLE
BUG: add_post_enctype added for every Uploadfield Instance

### ISSUE THREAD

**Issue by superdummy** [WebDevStudios/CMB2#232]
Hello, 

my `<head>` is filled with about ten times the following code (I use uploadfields): And its obvious, it should however only be included once!

```jQuery(document).ready(function(){
			$form = jQuery("#post");
			if ( $form.length ) {
				$form.attr( {
					"enctype" : "multipart/form-data",
					"encoding" : "multipart/form-data"
				} );
			}
		});
```

This sholuld roughly equal the nuber of times I use CMB2 upload fields in vaorius custom Post Types/Plugins. 
This code above is from the function `add_post_enctype()`. And looking it up further this should be added only once as it ist called via `$this->once( 'admin_head', array( $this, 'add_post_enctype' ) );`

This is the function that does the adding:
```
public function once( $action, $hook, $priority = 10, $accepted_args = 1 ) {
		$key = md5( serialize( func_get_args() ) );

		if ( in_array( $key, self::$hooks_completed ) ) {
			return;
		}

		self::$hooks_completed[] = $key;
		add_filter( $action, $hook, $priority, $accepted_args );
	}
```

But I was not able to track the problem down. 
Absolutely strangley  `func_get_args()`  seems to alway return false and then gets encoded to a different md5 hash each time. This seems to be the problem, that this md5 hash is different even that the imput seems to be "static" and is the same. Looking it up in the eclipse debugger `func_get_args()` alwasy returns false but nevertheless for each invocation the md5 hash is different. And then of course the hash can not be found in the "completed cache/array" and is added to the head multiple times. 

Could you please tell me what I should do next or please have a look yourself (I am quite new to php/wp...)

Thanks

BTW: Absolutely great plugin, love it!!!

**Comment by superdummy** [comment]
ok tracked it down further: 
1.) func_get_args() seems to be not debuggable with eclipse, alwas returns false. var_dump results in a different result: 
2.) func_get_args() indeed gets the complet Metabox object instance that is passed in as parameter in the $hook variable.  And this is of course different each time, as it is a complete different instance/metabox. 

So this method has a bug, as it always compares object identity. But here we also require a mode that only compares  if a method independent of the object identity has been added. 

One solution would be to add a further parameter "function_only" and if this is set, it will not consider the object identity that is passed in the $hook parameter but only the second array parameter, namely the function call itself (that here is add_post_enctype). And of couse $this->once( 'admin_head', array( $this, 'add_post_enctype' ) ) would then also have to be changed to pass in the new flag to not check the object identity. 

But problem is now clear...

**Comment by jtsternberg** [comment]
This is fixed (removed) in 3d85f3bfdc386930242d77f523ec75b3d25a20f1. That method is no longer needed since `file`/`file_list` use the WP media modal.
